web/controllers/user_controller.ex

```diff
@@ -41,6 +41,16 @@ defmodule Pairmotron.UserController do
     user = Repo.get!(User, id)
     changeset = User.changeset(user, user_params)
 
+    if current_assigned_user_id(conn) != id && changeset.changes[:password] do
```

Comment:
Since user is from `Repo.get!`, it should be safe to use `user.id` instead of `id` here. Then you don't have the `to_string` cast in `current_assigned_user_id`.